opencog/atomese/interpreter/logical_interpreter.h

```diff
@@ -148,5 +148,22 @@ bool logical_compare(const LinkValuePtr& p1, const LinkValuePtr& p2){
 
 	return std::equal(p1_value.begin(), p1_value.end(), p2_value.begin(), comparator);
 }
+
+/**
+ * compute logical_not.
+ * it expects the values of the LinkValues to be TRUE_LINK and FALSE_LINK
+ * @param LinkValuePtr&     p1
+ *
+ * @return                 LinkValue pointer containing the logical_not.
+ */
+LinkValuePtr logical_not(const LinkValuePtr& p) {
+	std::vector<ValuePtr> _result;
+	std::vector<ValuePtr> p_value = p->value();
+	std::vector<ValuePtr>::iterator it;
+	for(it = p_value.begin(); it != p_value.end(); ++it)
+		_result.push_back((HandleCast(*it)->get_type() == TRUE_LINK) ?
+		ValuePtr(createLink(FALSE_LINK)): ValuePtr(createLink(TRUE_LINK)));
```

Comment:
Ideally this should be static, especially since it in a rather heavily used part of the code.

@kasimebrahim , maybe we want to turn the locally static `TRUE_LINK` and `FALSE_LINK` values into something globally static, called `TRUE_LINK_VALUE`, etc.

Although ultimately I think we want to introduce some `BoolValue` type, but I guess in the meantime it makes sense (when that happen `TRUE_LINK_VALUE` would probably be rename `TRUE_VALUE`, but I suppose the static burden would still be needed).